File: src/main/python/lab999_functions/asinApp.py
"""
 asin function: inverse sine of a value in radians.

 @author rambabu.posa
"""
from pyspark.sql import SparkSession
from pyspark.sql import (functions as F)
import os

current_dir = os.path.dirname(__file__)
relative_path = "../../../../data/functions/trigo_arc.csv"
absolute_file_path = os.path.join(current_dir, relative_path)

# Creates a session on a local master
spark = SparkSession.builder.appName("asin function") \
    .master("local[*]").getOrCreate()

df = spark.read.format("csv") \
    .option("header", True) \
    .load(absolute_file_path)

df = df.withColumn("asin", F.asin(F.col("val"))) \
       .withColumn("asin_by_name", F.asin("val"))


df.show()

spark.stop()

File: src/main/java/net/jgp/books/spark/ch13/lab999_functions/AsinApp.java
package net.jgp.books.spark.ch13.lab999_functions;

import static org.apache.spark.sql.functions.asin;
import static org.apache.spark.sql.functions.col;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/**
 * asin function: inverse sine of a value in radians
 * 
 * @author jgp
 */
public class AsinApp {

  public static void main(String[] args) {
    AsinApp app = new AsinApp();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    SparkSession spark = SparkSession.builder()
        .appName("acos function")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("csv")
        .option("header", true)
        .load("data/functions/trigo_arc.csv");

    df = df.withColumn("asin", asin(col("val")));
    df = df.withColumn("asin_by_name", asin("val"));

    df.show();
  }
}
